### PR TITLE
removed powershell-yaml module dependency

### PR DESCRIPTION
**Pull request by yliaog**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
/kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
cleanup the dependency on powershell-yaml module

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: cluster/gce/windows/k8s-node-setup.psm1 [hunk 1]
```diff
@@ -147,6 +147,43 @@ function Dump-DebugInfoToConsole {
   } Catch { }
 }
 
+# Converts the kube-env string in Yaml
+#
+# Returns: a PowerShell Hashtable object containing the key-value pairs from
+#   kube-env.
+function ConvertFrom-Yaml-KubeEnv {
```
Comment (pjh) [inline cluster/gce/windows/k8s-node-setup.psm1]: Did you write this from scratch or is it based on something that already existed?

If the former, I'm a little wary of various corner cases popping up in the future. But I also have no idea about the quality of the powershell-yaml module we're already using.

**Comment by pjh** [comment]
> /test pull-kubernetes-e2e-windows-gce

Nice! Did the failed run above catch a real bug in your PR? I took a look through the [build-log](https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/84434/pull-kubernetes-e2e-windows-gce/1188686958780485632/build-log.txt), I see that the Windows nodes failed to join the cluster but I don't understand why the serial port output couldn't be fetched.

```
ERROR: (gcloud.compute.instances.get-serial-port-output) Could not fetch serial port output: The resource 'projects/k8s-jkns-e2e-gce-slow-1-2/zones/us-west1-b/instances/e2e-bb7cc1d52a-95646-windows-node-group-045s' was not found
```

File: cluster/gce/windows/k8s-node-setup.psm1 (hunk 1 above)
Comment (yliaog) [reply cluster/gce/windows/k8s-node-setup.psm1]: it is based on a code sample about text parsing in powershell. the yaml here is just 'key:value' pairs, we don't really need a full-blown yaml parser, IMO. also the kube env is provided by the control plane, e2e tests should be able to catch the corner cases before they hit production.

**Comment by yliaog** [comment]
yes, the pull test caught a real issue in my code, i fixed it, and the cluster could be created successfuly now (https://prow.k8s.io/log?id=1188947020870586368&job=pull-kubernetes-e2e-windows-gce)

it's likely the serial port output is not persisted.

**Comment by pjh** [comment]
> yes, the pull test caught a real issue in my code, i fixed it, and the cluster could be created successfuly now (https://prow.k8s.io/log?id=1188947020870586368&job=pull-kubernetes-e2e-windows-gce)

That's awesome, thanks for getting that presubmit working!!

**Comment by pjh** [comment]
priority/important-soon

**Pull request merged**